### DISCUSSION TITLE
Mesh refinement

### DISCUSSION THREAD

**Discussion by medhay36** [acroucher/PyTOUGH#45] [Q&A]
Good morning, 

I am trying to do a mesh refinement around a radioactive waste cell model to take into account interfaces and sealings.

![image](https://user-images.githubusercontent.com/122643918/226851151-99473ccd-eefc-4c35-8b6d-4f5b3399af50.png)

As described in PyTOUGH manual, refinement can be made providing a list of 2-D points.

I provided a list of 2d points describing borders of a rectangle but this doesn't work.

Has anyone tried this option especially when the mesh has plaid pattern  (plaid pattern are very rigid meshes and i always experienced problems in refinement) ?

**Comment by acroucher** [comment]
Are you wanting to refine your mesh in the horizontal or the vertical? If it's the mesh in the picture, I'm guessing you are probably wanting vertical refinement. The `mulgrid` `refine()` method only does horizontal refinement (i.e. refining the columns).

Note that the `mulgrid` geometry assumes a layer/column structure, so the only way to do vertical refinement is to introduce thinner layers (which will extend throughout the mesh). You can do this using the `refine_layers()` method, but this may not be what you had in mind.

**Reply by medhay36** [reply]
In fact, this was not what I had in mind. I wanted to make refinements in some particular areas (not a vertical refinement) 

But thank you very much dear professor for your answer. 
I continued with a plaid pattern mesh.
